File: CPP/ALL OTHER CODES/GeekForGeeks/palindrome.java
class plaindrome{

    public static void main(String[] args) {
        
        String s = "acaba";

        int start = 0;
        int end = s.length()-1;
        
        while(start<=end){
            if(s.charAt(start) != s.charAt(end)){
                System.out.println("Not palindrome");
                break;
            }
            start++;
            end--;
        }
        if(start>end) System.out.println("Palindrome");
        
    }
}

File: CPP/2020-2023/ALL OTHER CODES/COURSES/GeekForGeeks/palindrome.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    string s = "ababa";
    int start = 0; 
    int end = s.length() - 1;
    while(start<=end){
        if(s[start] != s[end]){
            cout << "Not Plindrome" << endl;
            break;
        }else{
            start++;
            end--;
        }
    }
    if(start>=end)cout << "Palindrome" << endl;
    
}